File: Lab-12/Text_Editor_210042141/src/EditorMemento.java
import java.io.Serializable;
import java.util.List;

public class EditorMemento implements Serializable
{
    private final String content;
    private final int cursorPosition;
    private final List<String> selections;

    public EditorMemento(String content, int cursorPosition, List<String> selections)
    {
        this.content = content;
        this.cursorPosition = cursorPosition;
        this.selections = selections;
    }

    public String getContent()
    {
        return content;
    }

    public int getCursorPosition()
    {
        return cursorPosition;
    }

    public List<String> getSelections()
    {
        return selections;
    }
}

File: Lab-12_Memento/EditorMemento.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_12_Memento
{
    public class EditorMemento
    {
        public string Content { get; }
        public int CursorPosition { get; }
        public List<string> Selections { get; }

        public EditorMemento(string content, int cursorPosition, List<string> selections)
        {
            Content = content;
            CursorPosition = cursorPosition;
            Selections = new List<string>(selections);
        }
    }
}
